controllers/mirrorpeersecret_controller.go

```diff
@@ -97,5 +102,34 @@ func mirrorPeerSecretReconcile(ctx context.Context, rc client.Client, req ctrl.R
 func (r *MirrorPeerSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Secret{}, builder.WithPredicates(utils.SourceOrDestinationPredicate)).
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(r.secretConfigMapFunc)).
 		Complete(r)
 }
+
+func (r *MirrorPeerSecretReconciler) secretConfigMapFunc(obj client.Object) []reconcile.Request {
+	ConfigMapRamenConfigKeyName := "ramen_manager_config.yaml"
+	var cm *corev1.ConfigMap
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return []reconcile.Request{}
+	}
+
+	ramenConfig := &ramendrv1alpha1.RamenConfig{}
+	err := yaml.Unmarshal([]byte(cm.Data[ConfigMapRamenConfigKeyName]), ramenConfig)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+
+	secrets := &corev1.SecretList{}
+	if err := r.Client.List(context.TODO(), secrets); err != nil {
+		return []reconcile.Request{}
+	}
```

Comment:
Does this list secret in all namespaces or just the current one?

Reply:
Some documentation says it is for all namespaces, some say its for default and few say current. 
So i passed list options with an empty Namespace field which will select all namespaces.